src/equations/linearized_euler_2d.jl

```diff
@@ -0,0 +1,184 @@
+# By default, Julia/LLVM does not use fused multiply-add operations (FMAs).
+# Since these FMAs can increase the performance of many numerical algorithms,
+# we need to opt-in explicitly.
+# See https://ranocha.de/blog/Optimizing_EC_Trixi for further details.
+@muladd begin
+
+
+@doc raw"""
+    LinearizedEulerEquations2D(rho, lambda)
+    
+The linearized Euler equations
+```math
+\partial t
+\begin{pmatrix}
+  v_1 \\ v_2 \\ p
+\end{pmatrix}
++
+\partial x
+\begin{pmatrix}
+\frac{1}{\rho} p \\ 0 \\ \lambda v1
+\end{pmatrix}
++
+\partial y
+\begin{pmatrix}
+0 \\ \frac{1}{\rho} p \\ \lambda v2
+\end{pmatrix}
+=
+\begin{pmatrix}
+0 \\ 0 \\ 0
+\end{pmatrix}
+```
+for an ideal acoustic wave with constant density ``\rho`` and second lame constant ``\lambda``.
```

Comment:
In a different project (the finished thesis of @lchristm) we use the linearized Euler equations with non-constant density. I am thus not sure how to proceed here: I do not want to stop this PR in its tracks, but I also do not believe that we need two versions of the linearized Euler equations, do we? Maybe just implement the non-constant density version here and "pay the price"? Or maybe @lchristm and @ttheine should get together to figure out which formulation of the LEE are suitable for both?

Reply:
Further, this implementation seems to be limited to zero mean velocity, or am I missing something?

Reply:
@sloede
> Further, this implementation seems to be limited to zero mean velocity, or am I missing something?
Yes, this is correct.

Where can we find the implementation of @lchristm? Or do you refer to the `Accoustic Perturbation Equations`? Sorry for my cluelessness. :)

Reply:
@lchristm has this in a private repo at the moment. I suggest @ttheine and him get together (e.g. on Slack) and hatch a plan on how to proceed

Reply:
Good idea!

Reply:
@jmark I implemented this form of the LEE:

$$\frac{\partial u}{\partial t} 
      + \frac{\partial f(u)}{\partial x} 
      + \frac{\partial g(u)}{\partial y}
      = 0$$

where

$$u = 
        \begin{pmatrix}
          \rho' \\ 
v_1' \\ 
v_2' \\ 
p'  
        \end{pmatrix}, \quad
      f(u) = 
        \begin{pmatrix}
          \bar{\rho} v_1' + \bar{v}_1 \rho' \\
          \bar{v}_1 v_1' + \frac{p'}{\bar{\rho}} \\
          \bar{v}_1 v_2' \\
          \bar{v}_1 p' + c^2 \bar{\rho} v_1'
        \end{pmatrix}, \quad
      g(u) = 
        \begin{pmatrix}
          \bar{\rho} v_2' + \bar{v}_2 \rho' \\
          \bar{v}_2 v_1' \\
          \bar{v}_2 v_2' + \frac{p'}{\bar{\rho}}\\
          \bar{v}_2 p' + c^2 \bar{\rho} v_2'
        \end{pmatrix},$$

$\bar{(\cdot)}$ denotes the uniform mean flow variables, $(\cdot)'$ denotes the acoustic quantities and $c$ is the speed of sound.

@ttheine Feel free to message me on the Trixi Slack to discuss how to proceed